docs/storage.md

```diff
@@ -0,0 +1,187 @@
+- Segment rotation, batching, and cleanup logic are identical to Ingestor.
+- The same enforcement logic for disk usage and segment count is available, but must be explicitly configured.
+
+## Configuration
+
+- **Collector supports the same storage-related configuration options as Ingestor:**
+  - `max-disk-usage`: Maximum total bytes allowed for all WAL segments (default: unlimited if unset or 0)
+  - `max-segment-count`: Maximum number of WAL segments allowed on disk (default: unlimited if unset or 0)
+  - `max-segment-size`: Maximum size per WAL segment
+  - `max-segment-age`: Maximum age per WAL segment
+  - `storage-dir`: Directory for WAL files
+- If `max-disk-usage` and `max-segment-count` are not set, Collector will not enforce any total storage limit and can fill the disk.
+
+## Enforcement and Behavior on Limit Exceedance
+
+- **When limits are set:**
+  - The WAL layer will reject new writes if adding data would exceed the configured disk usage or segment count.
+  - Collector will drop any telemetry data it cannot persist to the WAL due to these errors.
+  - There is no upstream backpressure: Collector does not signal Prometheus or other scrapers to slow down; it simply drops data that cannot be written.
+  - No HTTP 429 or similar error is sent to remote write clients; data loss is silent except for logs/metrics.
+- **When limits are not set:**
+  - Collector will continue writing WAL files until the disk is physically full, potentially causing node instability.
+
+## Recommendations
+
+- Always set `max-disk-usage` and/or `max-segment-count` in Collector's configuration to prevent disk exhaustion.
+- Monitor Collector logs and metrics for dropped data due to storage limits.
+
+## Summary Table: Collector vs Ingestor
+
+| Mechanism         | Ingestor Behavior                                    | Collector Behavior                |
+|-------------------|------------------------------------------------------|-----------------------------------|
+| WAL Write         | Durable, batched, WAL                                | Durable, batched, WAL             |
+| Limit Enforcement | Rejects writes, HTTP 429 to peers                    | Drops data, no upstream signal    |
+| Backpressure      | Propagates to senders/peers                          | No backpressure, silent data loss |
+| Config Defaults   | 10 GiB disk, 10,000 segments if unset                | Unlimited if unset                |
+
+---
+
+## Example: WAL Segment Lifecycle
+
+1. Telemetry arrives and is appended to the WAL segment for its table/schema.
+2. When the segment reaches max size or age, it is rotated.
+3. If the segment is small, it is transferred to a peer for coalescing; otherwise, it is uploaded directly to ADX.
+4. After upload/transfer, the segment is deleted from disk.
+5. If disk usage or segment count limits are exceeded, new writes/transfers are rejected until space is freed.
```

Comment:
This isn't right - currently we don't reject new writes as far as I know. If it is right, it is in conflict with the table above as we describe the behavior for backpressure.

At least at max segment counts, collector does stop prometheus scrapes but not sure if we consume this elsewhere as well.